### ISSUE TITLE
Add Maximize and Minimize buttons to the report windows

### ISSUE THREAD

**Issue by aleksejrs** [lune-stone/anki-addon-limit-new-by-young#16]

**Comment by lune-stone** [comment]
I'm not able to test this change as the buttons were already visible on my box before the change. Please let me know if it works for you.

**Comment by aleksejrs** [comment]
That added Maximize, but no Minimize for me.

**Comment by lune-stone** [comment]
Hmm, in that case can you run this will the debug console (`ctrl+shift+;`) and tell me the number on any of the dialog windows that has both buttons visible?

```
import aqt.qt as qt
import PyQt6

WindowType = PyQt6.QtCore.Qt.WindowType

for idx, value in enumerate([
        WindowType.WindowMinMaxButtonsHint,
        WindowType.WindowMinimizeButtonHint,
        WindowType.WindowMaximizeButtonHint,
        WindowType.WindowMinimizeButtonHint | WindowType.WindowMaximizeButtonHint,
        WindowType.WindowMinimizeButtonHint | WindowType.WindowMaximizeButtonHint | WindowType.WindowSystemMenuHint
    ]):
    dialog = qt.QDialog(mw)
    dialog.setWindowFlag(value)
    dialog.setWindowTitle(str(idx))
    dialog.setGeometry(0, 0, 200, 200)
    dialog.show()
```

**Comment by aleksejrs** [comment]
None of them has a minimize button, and all except for 1 have a maximize button. Because they are dialogs?

**Comment by lune-stone** [comment]
> Because they are dialogs?

Possibly, but I suspect it might also be the desktop environment or settings you are using. These window flags are really only hints to the desktop and it ultimately decides what to display. 

I tried installing Anki 24.04.1 Qt6  from the website to a Debian 12 with Gnome on a VM, but that also had min/max buttons before making any changes to the plugin (previously I was running KDE on Arch).

Can I get a bit more info about your setup? What version of the distro, and desktop is being used? How did you install Anki (apt, flatpak, source, etc.)?

Even if it isn't possible to fix this in the add-on code there is likely a configuration for the desktop that could be used as a workaround.

**Comment by aleksejrs** [comment]
Debian testing
Xfce (not necessarily fully updated)
anki-24.04.2-linux-qt6.tar.zst
This did add Minimize to AnkiMorphs, but it uses a `.ui` file: https://github.com/mortii/anki-morphs/commit/342255e739bbc628b9a25382f6418119a60929df

**Comment by lune-stone** [comment]
I was able to reproduce the issue on Xfce. 

After looking at the ui file you linked I thought using QMainWindow instead of QDialog would fix it, but with my testing it seems to behave the same (no buttons until you add the hint, then still missing the min button).

I didn't see any obvious settings in Xfce to show the controls either.

Going to have to look a bit more at the ui and think a bit on this one.

**Comment by lune-stone** [comment]
I started looking into this issue again. My last comment states that QMainWindow had the same behavior as QDialog, but that is not what I'm seeing now. It's not clear if my previous setup was different, or if I messed up in some way with my previous testing.

In any case. I've pushed a branch with what I think should be the fix if you would like to test it out and confirm that it is working for you.
